runtime/src/main/cpp/Porting.h

```diff
@@ -94,6 +99,23 @@ do {                          \
 } while(0)
 #endif
 
+template <typename F>
+// TODO: Use invoke_result without && when compiled with C++17
+ALWAYS_INLINE typename std::result_of<F&&()>::type tryOrTerminate(F&& tryClause) {
```

Comment:
It might make more sense to put this function to `Exceptions.h`.
Motivation -- the body relies on at least two implementation details of that module:
1. Kotlin exceptions are C++ exceptions.
2. `std::terminate` prints stack trace of "current" Kotlin exception.

Reply:
The same goes for the `TRY_CATCH` defined above I suppose?

Reply:
I'm not sure about this: `TRY_CATCH` is probably used only for C++ exceptions.

Reply:
I suggested above templated variant which is less bloated and does not imply extra try/catch, please consider.